tasks/apt/install.yml

```diff
@@ -0,0 +1,78 @@
+---
+
+- name: Ensure jenkins home dir is created
+  file:
+    path: "{{ jenkins_home }}"
+    mode: 0777
+    owner: "{{ jenkins_config_owner }}"
+    group: "{{ jenkins_config_group }}"
+    state: directory
+
+- name: Install apt PPA dependencies
+  apt:
+    name: python-software-properties
+
+- name: Install the Debian keyring
+  apt:
+    name: debian-archive-keyring
+
+- name: Add Debian experimental repository for OpenJDK
+  apt_repository:
+    repo: deb http://httpredir.debian.org/debian experimental main
+    state: present
+
+- name: Add Debian sid repository for OpenJDK
+  apt_repository:
+    repo: deb http://httpredir.debian.org/debian sid main
+    state: present
+
+- name: Create pinning file for apt
+  copy:
+    src: debian
```

Comment:
Let's put this into a `files/apt/` subdir (since it only applies to the apt method of installing) and rename it so that we know it contains the package pins, for example `files/apt/debian-package-pins`.